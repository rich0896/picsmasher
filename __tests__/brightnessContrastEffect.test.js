/* eslint-disable no-undef */
// __tests__/brightnessContrastEffect.test.js

const { BrightnessContrastEffect } = require('../effects.js'); // Adjust the path as needed
const { applyEffectAndGetImageData } = require('../helpers/testHelpers.js');

describe('BrightnessContrastEffect', () => {
    let ctx, canvas;

    beforeEach(() => {
        canvas = document.createElement('canvas');
        canvas.width = 10;
        canvas.height = 10;
        ctx = canvas.getContext('2d');

        // Fill the canvas with a solid color (e.g., rgb(100, 150, 200))
        ctx.fillStyle = 'rgb(100, 150, 200)';
        ctx.fillRect(0, 0, canvas.width, canvas.height);
    });

    test('should increase brightness correctly', () => {
        const effect = new BrightnessContrastEffect({ brightness: 50, contrast: 0 });
        const serializedImageData = applyEffectAndGetImageData(effect, ctx, canvas);

        expect(serializedImageData).toMatchSnapshot();
    });

    test('should decrease brightness correctly', () => {
        const effect = new BrightnessContrastEffect({ brightness: -50, contrast: 0 });
        const serializedImageData = applyEffectAndGetImageData(effect, ctx, canvas);

        expect(serializedImageData).toMatchSnapshot();
    });

    test('should increase contrast correctly', () => {
        const effect = new BrightnessContrastEffect({ brightness: 0, contrast: 50 });
        const serializedImageData = applyEffectAndGetImageData(effect, ctx, canvas);

        expect(serializedImageData).toMatchSnapshot();
    });

    test('should handle combined brightness and contrast correctly', () => {
        const effect = new BrightnessContrastEffect({ brightness: 20, contrast: -30 });
        const serializedImageData = applyEffectAndGetImageData(effect, ctx, canvas);

        expect(serializedImageData).toMatchSnapshot();
    });
});
