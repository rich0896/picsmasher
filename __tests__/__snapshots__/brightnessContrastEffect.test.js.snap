// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrightnessContrastEffect should decrease brightness correctly 1`] = `ImageData(10, 10, [50, 100, 150, 255, 50, 100, 150, 255, 50, 100...])`;

exports[`BrightnessContrastEffect should handle combined brightness and contrast correctly 1`] = `ImageData(10, 10, [126, 165, 205, 255, 126, 165, 205, 255, 126, 165...])`;

exports[`BrightnessContrastEffect should increase brightness correctly 1`] = `ImageData(10, 10, [150, 200, 250, 255, 150, 200, 250, 255, 150, 200...])`;

exports[`BrightnessContrastEffect should increase contrast correctly 1`] = `ImageData(10, 10, [86, 161, 235, 255, 86, 161, 235, 255, 86, 161...])`;
